src/clients/FtpRelay.cc

```diff
@@ -385,9 +385,16 @@ Ftp::Relay::forwardReply()
     reply->sources |= Http::Message::srcFtp;
 
     setVirginReply(reply);
-    markParsedVirginReplyAsWhole("Ftp::Relay::handleControlReply() does not forward partial replies");
```

Comment:
I see nothing wrong with this _call_: We got a whole virgin response here, right? If so, we should call markParsedVirginReplyAsWhole().

I wonder if the bug is in markParsedVirginReplyAsWhole() implementation: The method checks startedAdaptation, but that flag does not cover adaptationAccessCheckPending state. Can we fix Bug 5279 by checking  !doneWithAdaptation() in markParsedVirginReplyAsWhole() instead, keeping FTP code intact?

If yes, please check other places where we check adaptationAccessCheckPending, doneWithAdaptation(), or startedAdaptation. Do we check the correct state in every such place?